dap_prinz_green_jobs/pipeline/ojo_application/ojo_sample/sample_ojo.py

```diff
@@ -69,33 +75,87 @@ def filter_data(data, filter_ids_set, id_col_name="id"):
     )
     random.shuffle(mixed_sample_ids)
 
+    # SAMPLE FOR LARGE SAMPLE
+    logger.info("Getting large sample")
+    job_title_data = pd.read_parquet(config["ojo_s3_file_adverts_ojd_daps_extract"])
+    # There are duplicate rows with different date formats
+    job_title_data = job_title_data.drop_duplicates(
+        subset=job_title_data.columns.difference(["created"])
+    )
+    # load job locations
+    locations_data = pd.read_parquet(config["ojo_s3_file_locations"])
+    locations_data = locations_data.drop_duplicates()
+
+    jobtitles2soc = load_s3_data(
+        BUCKET_NAME,
+        f"outputs/data/ojo_application/deduplicated_sample/jobtitles2soc4_production_{production}.json",
+    )
+
+    job_title_locations = (
+        pd.merge(job_title_data, locations_data, on="id")
+        .query("id in @deduplicated_ids_list")
+        .query("is_uk == 1")
+        .query("is_large_geo == 0")[
+            [
+                "id",
+                "job_title_raw",
+                "itl_2_code",
+                "itl_2_name",
+                "itl_3_code",
+                "itl_3_name",
+            ]
+        ]
+        .reset_index(drop=True)
+    )
+
+    job_title_locations["soc4_code"] = job_title_locations.job_title_raw.map(
+        jobtitles2soc
+    )
+    job_title_locations["soc4_code_itl2_code"] = (
+        job_title_locations.soc4_code + "_" + job_title_locations.itl_3_code
+    )
+
+    soc4_itl2_weights = (
+        job_title_locations.groupby(["soc4_code_itl2_code"]).size()
+        / job_title_locations.shape[0]
+    )
+    job_title_locations[
+        "soc4_code_itl2_code"
+    ] = job_title_locations.soc4_code_itl2_code.map(soc4_itl2_weights)
+
+    large_sample_df = job_title_locations.sample(
+        n=desired_sample_size, weights="soc4_code_itl2_code", random_state=random_seed
+    )
```

Comment:
this is how i sampled by soc4 codes and itl2 codes i.e. passed the combinations as weights - this is what i'd focus on in the review!